### DISCUSSION TITLE
Table width as in tabulary package?

### DISCUSSION THREAD

**Discussion by fredrikpaues** [lvjr/tabularray#397] [Q&A]
Is it possible with `tabularray` to set table width as is done with `tabulary`? That is, if the natural width of the table is less than a supplied maximum width, then the table is typeset with its natural width. But if the natural width of the table is greater than the supplied maximum width, then the table is typeset as a `tabularx` table with the supplied maximum width as its width.

**Comment by lvjr** [comment]
I have added a comment to issue #365.

**Comment by note286** [comment]
`Q[co=-1,c,m]` or `X[-1,c,m]` is your need, i.e.,negative number for `co` . Such as you can use `X[-2,c,m]X[-1,c,m]`.

**Reply by fredrikpaues** [reply]
I can't seem to get it to work. I have posted [a question over on TeXSE](https://tex.stackexchange.com/questions/683538/replicate-tabularys-l-column-with-tabularray)

**Reply by LumosX** [reply]
Not certain I understand the question in the OP completely, but I figured I'd post here for posterity, as the negative coefficients worked for me. In my use case (the tables are always of standard width, and I was interested in tinkering with column widths only), I've got tables generated at compile-time, most of the form:

| Left | Right |
|:--- |:--|
| short text | short text |
| short text | short text |
| very very long text | short text |

Using a `colspec` of `{Xl}` did a fine job for most of them that looked like the above, but a rarer case, where we had:

| Left | Right |
|:--- |:--|
| very very long text | short text |
| short text | very very long text |

... would look absolutely horrid. The width of the table in a `twocolumn` page would be insufficient to host both columns, which would lead to the left-hand column wrapping (naturally, being the `X` one) and the right-hand one looking oversized. I was desperate to find a way to make the right-hand column as small as possible, and have it wrap if it exceeds a certain threshold, but since tabularray does not support `V`-type columns, I needed an alternative. Negative coefficients turned out to do exactly what I needed, and setting `colspec={XX[l,-1]}` fixed my issue completely: now the right-hand column remains small, and wraps if it touches the left-hand one.

Each time I contemplate opening an issue about this package, it turns out there's a way to fix it. Must be proof it's a good project, not to mention it's been a life-safer for certain scenarios. :)
